tests/integration/test_manifest_generation.py

```diff
@@ -0,0 +1,503 @@
+noted in the test function docstrings.
+"""
+import os
+from io import BytesIO
+
+import requests
+from openpyxl import load_workbook
+
+LIGHT_BLUE = "FFEAF7F9"  # Required cell
+GRAY = "FFE0E0E0"  # Header cell
+WHITE = "00000000"  # Optional cell
+
+
+class TestManifestGeneration:
+    def test_single_manifest_generation_excel(
+        self, manual_test_verification_path: str
+    ) -> None:
+        """
+        Download a manifest from the Schematic API and verify that it is a valid Excel
+        file. We are validating the following:
+
+        - The first row of the Excel file contains the column headers
+        - The first row is locked on scroll
+        - Each cell in the first row has a comment "TBD"
+        - The cell B2 in Sheet1 has a dropdown list with values from Sheet2!B2:B4
+        - The cell D2 in Sheet1 has a dropdown list with values from Sheet2!D2:D3
+        - The cell F2 in Sheet1 has a dropdown list with values from Sheet2!F2:F6
+        - The workbook contains two sheets: "Sheet1" and "Sheet2"
+        - "Sheet2" is hidden
+        - The values in "Sheet2" are as expected
+        - The fill colors of the first row cells are as expected
+        - Conditional formatting is functioning as expected
+
+
+        Manual verification steps:
+
+        - Open the Excel file prefixed with TestManifestGeneration_test_single_manifest_generation_excel.xlsx
+        - When Diagnosis = "Cancer", [Cancer Type, Family History] is Light Blue (Required)
+        """
+        # GIVEN a valid example manifest to generate
+        # TODO: Determine if we are able to use this test for both integration tests running through pytest when the API server is running, or when the API server is passed as a fixture
+        url = "http://localhost:3001/v1/manifest/generate"
+        params = {
+            "schema_url": "https://raw.githubusercontent.com/Sage-Bionetworks/schematic/develop/tests/data/example.model.jsonld",
+            "title": "Example",
+            "data_type": "Patient",
+            "use_annotations": "false",
+            "output_format": "excel",
+            "strict_validation": "true",
+            "data_model_labels": "class_label",
+        }
+        headers = {
+            "accept": "application/json"
+            # TODO: Include an authorization header if required
+        }
+
+        # WHEN we make a request to the Schematic API
+        response = requests.get(url, headers=headers, params=params)
+
+        # THEN we expect a successful response
+        assert response.status_code == 200
+
+        # Load the response content into memory
+        content = BytesIO(response.content)
+        workbook = load_workbook(content)
+        sheet1 = workbook["Sheet1"]
+
+        # Track column positions
+        columns = {cell.value: cell.column_letter for cell in sheet1[1]}
+
+        # AND the content of the first sheet is as expected
+        assert sheet1[f"{columns['Patient ID']}1"].value == "Patient ID"
```

Comment:
@BryanFauble Rather than having all of these assert statements with manually entered values, would it make more sense to use the provided example files and directly compare the generated manifest file to the example where possible? Like
```
    for col in test_sheet.iter_cols(1, test_sheet.max_column):
        for cell in col:
            assert cell.value == example_sheet[cell.coordinate].value
```

Reply:
Let's consider a few possibilities:

1. Example files stored outside of source control are the "Source of truth" for the tests that are running. If those example files are modified and tests are not updated "In lockstep" then the tests are going to fail for some amount of time.
2. Example files stored inside source control, as example files are updated, so too, are the tests that are using them.
3. There are no example files stored anywhere and we are doing the verbose verification of the content of those files like what is being done here.

@linglp @andrewelamb @GiaJordan I would like you opinion on the direction that we should take here.

Reply:
Something else to think about as well is that the final file to compare against is not the file as it is downloaded from the schematic API. It is the state of the file after 1 or more manipulations have been done to it (See the case where the test plan mentions changing drop downs to verify conditional formatting).

Those change would need to be made in the code in order to have the "same" file to compare.

This is one of the things that is point me towards handling all of the verification in the code so we have 1 single spot to maintain everything. Yes, it is a bit more maintenance, but it's nothing that an LLM cannot do for us.

Reply:
Sorry, I must have missed this when I first got tagged. My preference would be to use a "source of truth" file to test all cell values. Something like

```
test_df = pnadas.read_excel(response.content) # not sure of pandas cna do this
expected_df = pandas.read_csv("expected.csv")

for col in expected_df.columns:
  assert expected_df[col].values.to_list() == test_df[col].values.to_list()
```
But it's nto a strong perefence and doesn't need to be adressed in this PR.